src/net.cpp

```diff
@@ -2820,12 +2820,96 @@ bool CConnman::ForNode(NodeId id, std::function<bool(CNode* pnode)> func)
     for (auto&& pnode : vNodes) {
         if(pnode->GetId() == id) {
             found = pnode;
+            found->AddRef();
             break;
         }
     }
-    return found != nullptr && NodeFullyConnected(found) && func(found);
+    bool res = found != nullptr && NodeFullyConnected(found) && func(found);
+    found->Release();
+    return res;
 }
 
+void CConnman::ForEachNode(std::function<void (CNode* pnode)> func)
```

Comment:
I mean it does, but I'm not too worried about requiring move for a callable type (places where we std::move a lambda into this will still get the benifit of calling std::function(Callable&&)). The real concern here, of course, is that it introduces a performance penalty, but its not huge (an extra function call and few if's on top of the callback function call) and we're already calling this with lambda functions, so I dont think we were ever too worried about the performance here.